app/models/OcrImport.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+class OcrImport {
+
+    public function tesseract($data){
+        $url = getenv('OCR_URL');
+        
+        //POST_PARAMS : image URI & engine, in json type
+        $datajson = array("img_url" => URL::to('/').'/'.$data['img_url'], 
+                      "engine" => getenv('OCR_ENGINE'),
+                      "lang" => Language::find(1)->codeiso
+                     );
+        $postParams = json_encode($datajson);
+        \Log::debug('OCR postParams :' . $postParams);
+        
+        // Set HTTP POST Request+Params
+        $ch = curl_init();
+        curl_setopt($ch, CURLOPT_URL, $url);
+        curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "POST");
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        curl_setopt($ch, CURLOPT_POSTFIELDS, $postParams);
+        curl_setopt($ch, CURLOPT_HTTPHEADER, array(
+            'Content-Type: application/json',
+            'Content-Length: ' . strlen($postParams))
+        );
+        
+        $result = curl_exec($ch);
+        \Log::info('OCR strResponse:' . $result);
+        
+        // Close the Curl Session.
+        curl_close($ch);
+        
+        return $result;
+    }
+    
+    public function fire($job,$data){
+        //OCR URL Determination
+        if( getenv('OCR_ENGINE') == 'tesseract'){
+            $text = OcrImport::tesseract($data);
+        }
+        else{/*TODO FIX : no other OCR platform for now, abort the detection*/
+            $job-> delete();//delete jobs from the queue
+            return;
+        }
```

Comment:
Seems ok for me, but please replace by a switch case if possible (see comment above)